testkit/src/api.rs

```diff
@@ -192,16 +225,14 @@ impl TestKitApi {
     /// # Panics
     ///
     /// - Panics if the response has a non-40x response status.
-    pub fn get_err<D>(&self, kind: ApiKind, endpoint: &str) -> D
-    where
-        for<'de> D: Deserialize<'de>,
-    {
-        TestKitApi::get_internal(
-            &self.public_handler,
-            &format!("{}/{}", kind.into_prefix(), endpoint),
-            true,
-            true,
-        )
+    pub fn get_err(&self, kind: ApiKind, endpoint: &str) -> ApiError {
+        let url = format!("http://localhost:3000/{}/{}", kind.into_prefix(), endpoint);
+        let response = match request::get(&url, Headers::new(), &self.public_handler) {
+            Ok(response) |
```

Comment:
I *think* I've written a line comment here, but somehow it got lost :thinking: It seems that the new version of `get_err` will actually attempt to create errors from non-erroneous (i.e., 20x) responses (which would differ from the current behavior: panicking if the response code is not erroneous; I'd argue that this is reasonable behavior). Could you add a test to fix that this is not the case?

Reply:
@slowli, so, have I got you right? We should panic, if there's not erroneous code - I should preserve this behavior and write tests for it.

Reply:
Yeah, that describes my intention.